### DISCUSSION TITLE
Reverse Proxy Management

### DISCUSSION THREAD

**Discussion by swissbyte** [coollabsio/coolify#2558] [Feature Requests]
Hello

im running coolify inside an proxmox environment. Coolify is handling all the incomming traffic on 80 and 443.

But there are still some services that i have to host on proxmox itself and not as a container part of coolify.

it would be great if we could just setup some custom reverse proxy settings to revers proxy traffic for services outside of coolify.

But coolify should still handle the ssl termination with letsencrypt.

atm im doing this with nginx proxy manager inside a container that runs on coolify. It works but its a bit overhead. 

Thx for this awesome piece of software.

**Comment by andreitere** [comment]
Found myself in the exact same spot. Built a little [something](https://github.com/andreitere/coolify-nginx-reverse-proxy). Just need to setup an env var to point it to the host:port I need to reverse proxy. 

Definitely would be nicer to have something directly in Coolify to configure stuff like this

**Reply by gltched-usr** [reply]
@andreitere Maybe I'm doing something wrong but I get this error:
```
[2024-Jul-03 16:26:27.381044] 2024/07/03 16:25:48 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
2024/07/03 16:25:51 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
2024/07/03 16:25:53 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
2024/07/03 16:25:56 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
2024/07/03 16:25:59 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
2024/07/03 16:26:07 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
2024/07/03 16:26:16 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
2024/07/03 16:26:26 [emerg] 1#1: invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
nginx: [emerg] invalid number of arguments in "proxy_set_header" directive in /etc/nginx/nginx.conf:12
```

I set `PROXY_HOST_PORT` env to `proxy_manager_app:80`. I've made sure that nginx proxy manager and the coolify container are in the same network (coolify).

**Reply by andreitere** [reply]
@gltched-usr Can't really reproduce it. Did you build it yourself (the image) or you are using this one: ghcr.io/andreitere/coolify-nginx-reverse-proxy?

**Reply by gltched-usr** [reply]
I cloned your [github repo](https://github.com/andreitere/coolify-nginx-reverse-proxy) in coolify, set the env var,  added nginx proxy manager to the coolify network and deployed it

**Reply by andreitere** [reply]
Did some updates. You may want to try again! Hope it works this time. Also there’s a section in the docs mentioning that when connecting to the predefined network you will have to append the uuid to the host (https://coolify.io/docs/knowledge-base/docker/compose)

**Reply by gltched-usr** [reply]
@andreitere now the container is deploying but I'm not sure what do enter for the domain to make the proxy work. I tried entering domains I have registerd in nginx proxy manager but this only results in a to many redirects error.

**Reply by andreitere** [reply]
that might be too many proxies in the chain. why not directly setting up in this the upstream you have configured in nginx pm? :)

**Reply by gltched-usr** [reply]
There are only two proxys in the chain. Coolify and nginx pm. 

Setting up the upstream directly to the target container means I need to run your tool for every target proxy host I have? If so I don't think thats a good solution, I have like arround 20 proxys configured in nginx pm.

**Reply by lesimoes** [reply]
Works like charm for me!

**Comment by taqqi1214** [comment]
https://freddydumont.com/blog/npm-custom-locations
